deepspeed/inference/engine.py

```diff
@@ -86,6 +86,27 @@ def build_bloom_alibi_tensor(attention_mask: torch.Tensor, num_heads: int, dtype
         return alibi.reshape(batch_size * num_heads, 1, seq_length).to(dtype)
 
 
+from typing import Optional
+
+
+def build_mpt_atten_bias_tensor(self,
+                                device,
+                                dtype,
+                                attention_mask: Optional[torch.ByteTensor] = None,
+                                prefix_mask: Optional[torch.ByteTensor] = None,
+                                sequence_id: Optional[torch.LongTensor] = None):
+    (attn_bias, attention_mask) = self._attn_bias_orig(device,
+                                                       dtype,
+                                                       attention_mask=attention_mask,
+                                                       prefix_mask=prefix_mask,
+                                                       sequence_id=sequence_id)
+    if dist.is_initialized():
```

Comment:
that's a nice trick to make it MP-compatible. Thanks :) A minor comment here is to maybe move this function as well as the `build_alibi_...` to a a different file as utility for injection when auto-tp is used.

Reply:
I do not know where I should place the function. Any suggestion?